src/Controller/Component/AuthComponent.php

```diff
@@ -403,10 +406,10 @@ protected function _isLoginAction(Controller $controller)
      * Handle unauthorized access attempt
      *
      * @param \Cake\Controller\Controller $controller A reference to the controller object
-     * @return \Cake\Http\Response
+     * @return \Cake\Http\Response|null
```

Comment:
Must be reverted.

Reply:
`Controller::redirect` could return `null` so this should be fine

Reply:
Why can it? It should either return a response or throw an exception, right?

Reply:
Because of `Controller::redirect():583` 

```
if ($event->isStopped()) {
  return null;
}
```

Reply:
OK